### DISCUSSION TITLE
Only temperature on dashboard

### DISCUSSION THREAD

**Discussion by jbonnet69** [mp-se/fermentrack-docker#37] [Q&A]
Hello @mp-se,
I added my new ispindel to Fermentrack:

![image](https://user-images.githubusercontent.com/85406128/121788166-719f5880-cbcb-11eb-8da1-9c966bbd2b8a.png)

I created a test batch and keep the ispindel in my kitchen to test.
I see the temperature and the battery info, but:
- no gravity info
- no graphic on dashboard

![image](https://user-images.githubusercontent.com/85406128/121788150-46b50480-cbcb-11eb-843b-f7409e71d720.png)

And when I want to "view full csv" I have the following error. 
![image](https://user-images.githubusercontent.com/85406128/121788154-57657a80-cbcb-11eb-8aaf-aa55e0c2dfe3.png)
![image](https://user-images.githubusercontent.com/85406128/121788158-5f251f00-cbcb-11eb-8114-65407d6786e1.png)

Do you have an idea on these problems?

Thanks in advance

**Comment by mp-se** [comment]
Based on your description it looks like a directory access issue with the /data mount point. This is where fermetrack stores the log files for fermentration. The graph reads this file and renders the graph so if this is not accessible it will not show anything.

I would suggest that you check the access rights from within the container by running a few linux commands;

How you attach to the container will depend on your runtime but if you are running standard docker on linux this command will give you a command prompt:

>docker exec -it fermentrack /bin/bash

 To check the permissions run this command;

>ls -al /home/fermentrack/fermentrack/

Check the permission on the folder called data, it should be drwxrwxrwx with the user fermentrack:fermentrack as the owner. 

You can also check the permissions of the files in the data directory

>ls -al /home/fermentrack/fermentrack/data

The other thing you can look at is the ngnix logs which you can view the last entries with these commands;

>tail -20 /var/log/nginx/error.log
>tail -20 /var/log/nginx/access.log

This might give you some insight to why you get 403 error. 

I'm running the latest version on my brewery without any issues. The image should also make sure that the access to directory is correct or it would not start.

Please share any log entries that could share some insights to what the problem is.....

**Reply by jbonnet69** [reply]
Thank you.

Here are the results of the different commands:
> ls -al /home/fermentrack/fermentrack/

Result : 
drwx--x--x  2 fermentrack fermentrack 4096 Jun 12 20:24 data

> ls -al /home/fermentrack/fermentrack/data

Result :
root@5a47162b1c55:/home/fermentrack# ls -al /home/fermentrack/fermentrack/data
total 440
drwx--x--x  2 fermentrack fermentrack   4096 Jun 12 20:24  .
drwxr-xr-x 31 fermentrack fermentrack   4096 Jun  8 01:16  ..
-rwx--x--x  1 fermentrack fermentrack   3597 Jun 12 20:16 'Gravity Device 1 - L2 - Test_annotations.almost_json'
-rwx--x--x  1 fermentrack fermentrack   2674 Jun 12 20:16 'Gravity Device 1 - L2 - Test_full.csv'
-rwx--x--x  1 fermentrack fermentrack   1353 Jun 12 20:16 'Gravity Device 1 - L2 - Test_graph.csv'
-rwx--x--x  1 fermentrack fermentrack 197858 Jun 13 06:55 'Gravity Device 2 - L3 - Test_annotations.almost_json'
-rwx--x--x  1 fermentrack fermentrack 141535 Jun 13 06:55 'Gravity Device 2 - L3 - Test_full.csv'
-rwx--x--x  1 fermentrack fermentrack  73059 Jun 13 06:55 'Gravity Device 2 - L3 - Test_graph.csv'

> tail -20 /var/log/nginx/error.log

Result:
root@5a47162b1c55:/home/fermentrack# tail -20 /var/log/nginx/error.log
2021/06/12 20:12:11 [error] 35#35: *14256 open() "/home/fermentrack/fermentrack/data//Gravity Device 1 - L2 - Test_full.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%201%20-%20L2%20-%20Test_full.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/logs/"
2021/06/12 20:13:00 [error] 35#35: *14256 open() "/home/fermentrack/fermentrack/data//Gravity Device 1 - L2 - Test_full.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%201%20-%20L2%20-%20Test_full.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/logs/"
2021/06/12 20:13:23 [error] 36#36: *14268 open() "/home/fermentrack/fermentrack/data//Gravity Device 1 - L2 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%201%20-%20L2%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/1/"
2021/06/12 20:16:33 [error] 35#35: *14340 open() "/home/fermentrack/fermentrack/data//Gravity Device 1 - L2 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%201%20-%20L2%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/1/"
2021/06/12 20:17:06 [error] 35#35: *14340 open() "/home/fermentrack/fermentrack/data//Gravity Device 1 - L2 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%201%20-%20L2%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/1/"
2021/06/12 20:17:20 [error] 35#35: *14340 open() "/home/fermentrack/fermentrack/data//Gravity Device 1 - L2 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%201%20-%20L2%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/1/"
2021/06/12 20:26:37 [error] 35#35: *14497 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:32:14 [error] 35#35: *14625 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:33:43 [error] 36#36: *14652 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/log/3/view/"
2021/06/12 20:47:01 [error] 35#35: *14802 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:47:08 [error] 35#35: *14802 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:47:09 [error] 36#36: *14799 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:47:10 [error] 35#35: *14802 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:47:10 [error] 36#36: *14799 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:47:18 [error] 35#35: *14802 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:47:21 [error] 36#36: *14799 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:47:29 [error] 36#36: *14799 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/12 20:53:42 [error] 35#35: *15001 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"
2021/06/13 15:57:58 [warn] 17#17: the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:1
2021/06/13 15:58:38 [error] 34#34: *9 open() "/home/fermentrack/fermentrack/data//Gravity Device 2 - L3 - Test_graph.csv" failed (13: Permission denied), client: ::ffff:172.18.0.1, server: _, request: "GET /data/Gravity%20Device%202%20-%20L3%20-%20Test_graph.csv HTTP/1.1", host: "192.168.1.22:60", referrer: "http://192.168.1.22:60/gravity/sensor/2/"

> tail -20 /var/log/nginx/access.log

Result:
root@5a47162b1c55:/home/fermentrack# tail -20 /var/log/nginx/access.log
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:29 +0000] "GET / HTTP/1.1" 200 2139 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:29 +0000] "GET /api/lcd/ HTTP/1.1" 200 2 "http://192.168.1.22:60/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:29 +0000] "GET /api/gravity/ HTTP/1.1" 200 372 "http://192.168.1.22:60/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:34 +0000] "GET /api/lcd/ HTTP/1.1" 200 2 "http://192.168.1.22:60/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:34 +0000] "GET / HTTP/1.1" 200 2139 "http://192.168.1.22:60/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:34 +0000] "GET /api/lcd/ HTTP/1.1" 200 2 "http://192.168.1.22:60/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:34 +0000] "GET /api/gravity/ HTTP/1.1" 200 372 "http://192.168.1.22:60/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:38 +0000] "GET /gravity/sensor/2/ HTTP/1.1" 200 7025 "http://192.168.1.22:60/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:38 +0000] "GET /api/gravity/2/ HTTP/1.1" 200 372 "http://192.168.1.22:60/gravity/sensor/2/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:172.18.0.1 - - [13/Jun/2021:15:58:38 +0000] "GET /api/gravity/ispindel/2/ HTTP/1.1" 200 235 "http://192.168.1.22:60/gravity/sensor/2/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36"
::ffff:127.0.0.1 - - [13/Jun/2021:15:58:58 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"
::ffff:127.0.0.1 - - [13/Jun/2021:15:59:58 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"
::ffff:127.0.0.1 - - [13/Jun/2021:16:00:59 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"
::ffff:127.0.0.1 - - [13/Jun/2021:16:01:59 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"
::ffff:127.0.0.1 - - [13/Jun/2021:16:02:59 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"
::ffff:127.0.0.1 - - [13/Jun/2021:16:04:00 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"
::ffff:127.0.0.1 - - [13/Jun/2021:16:05:00 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"
::ffff:127.0.0.1 - - [13/Jun/2021:16:06:00 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"
::ffff:127.0.0.1 - - [13/Jun/2021:16:07:00 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"
::ffff:127.0.0.1 - - [13/Jun/2021:16:08:01 +0000] "GET / HTTP/1.1" 200 8773 "-" "curl/7.64.0"


The log in the graphic interface of the Docker of Synology is:
![image](https://user-images.githubusercontent.com/85406128/121814746-f6917d00-cc72-11eb-9ae2-7b309680b09d.png)
![image](https://user-images.githubusercontent.com/85406128/121814754-0315d580-cc73-11eb-9d01-ec624a2348d5.png)

**Reply by mp-se** [reply]
Can you check what access rights there are in the folders in the synology gui. The fermentrack user in the container does not have access to the mounted volumes.

**Reply by jbonnet69** [reply]
the container is launched by "tiphju" user which is the owner of the folder fermentrack:
![image](https://user-images.githubusercontent.com/85406128/121818249-46c60a80-cc86-11eb-928b-6391fe9c1b42.png)

And this user has a total control on the folder:
![image](https://user-images.githubusercontent.com/85406128/121818269-66f5c980-cc86-11eb-9469-81d9ec4285c8.png)


Should I create a specific user for fermentrack?

**Reply by mp-se** [reply]
Linux does not care about the name of the user only the UID of that user. Fermentrack has UID 2001 and a user on the host system with that UID needs to have RW access to the folder. You can also grant users of a group with GID 2000 access to that folder. 

You have two options, either create a user with that UID and give that permission to the folder or grant all users RW access to the folder. 

I dont think you can specify UID in the synology GUI so you will need to do an ssh session and use the commandline to create a user with a specifed UID:

**Reply by jbonnet69** [reply]
I gave all rights to "everyone" to the fermentrack folder: I have now a graph and access to the log.
But still no gravity.
My ispindel is in the water now and:
![image](https://user-images.githubusercontent.com/85406128/121849789-16b15280-ccec-11eb-8bc8-77ad6e0701ad.png)

**Reply by mp-se** [reply]
Have you done a calibration in fermentrack for the device ? Fermentrack calculates the gravity based on its own formula and the reported tilt.

**Reply by jbonnet69** [reply]
The calibration was done by my seller in the ispindel config.
I just copied it in Fermentrack: it works.
Thank you again!
